maintainers/scripts/activity/activity.bb

```diff
@@ -0,0 +1,425 @@
+#!/usr/bin/env bb
```

Comment:
```suggestion
#!/usr/bin/env nix-shell
#!nix-shell -p babashka curl jq -i bb
```

Reply:
This adds 1-2 seconds to the script startup, so no

Reply:
... or maybe once the script has a use case and is stable. But I think there may be better ways to set the interpreter, e.b. by running it through fixup phase ...

Reply:
You can still call the script with `bb ./script.bb` for those that know about Babashka, but assuming that people know what Babashka is to use this script is bad for other maintainers that are new to the project.

`nix-shell` is something universal that works everywhere.

Reply:
I know.

In fact, I've already had the nix-shell shebang. I got rid of it again, because it's inconvenient at this stage of development and makes a bad first impression (arguably worse than having to nix-shell into a thing).

Also, frankly, at this point in time, I have no interest in this level of improvement. What I'm looking for, right now, are substantial workflows to be implemented, that can actually enable use cases. IOW, the shebang won't make or break the concept, and we are very much in the concept stage right now, where we need to find out if this thing is worth having at all. The shebang is going on the backlog. Please respect.

Reply:
> I know.
> 
> In fact, I've already had the nix-shell shebang. I got rid of it again, because it's inconvenient at this stage of development and makes a bad first impression (arguably worse than having to nix-shell into a thing).
> 
> Also, frankly, at this point in time, I have no interest in this level of improvement. What I'm looking for, right now, are substantial workflows to be implemented, that can actually enable use cases. IOW, the shebang won't make or break the concept, and we are very much in the concept stage right now, where we need to find out if this thing is worth having at all. The shebang is going on the backlog. Please respect.

Where did I disrespected you? I think you didn't gave a good explanation before marking my issue as "solved", this is why I marked it as unsolved, but it is not to disrespected your work.

There is at least one other reviewer that concur to me that `nix-shell` would be a good thing to have here (see the reaction in the original comment).

Reply:
But ok, fine, I am out of reviewing this PR. I am unfollowing this discussion, please do not ping me again.

Reply:
(For anyone that thinks this is an overreaction, see PR https://github.com/NixOS/nixpkgs/pull/337478, and if you want to see an opinion about a third party, see https://github.com/NixOS/nixpkgs/pull/337478#issuecomment-2311319721).

I am out, feel free to "resolve" this conversation for it to not appear to other reviewers.

Reply:
I didn't say you disrespected me. Though you are coming close. Again.

I asked you to respect that my decision to have the shebang to my liking during the draft phase, while I'm working on it.

I also asked you to respect what kind of feedback I was soliciting at this point. I thought I was very specific about this from the start, and nitpicking the shebang was decidedly not among it.

If you decide to come back, please lay off this or get muted. Bye.

Reply:
> I asked you to respect that my decision to have the shebang to my liking during the draft phase, while I'm working on it.

I am not saying you need to solve this right now, but please keep the issue open if this is not solved but planned.

> If you decide to come back, please lay off this or get muted. Bye.

This is one of the most toxic replies I got in nixpkgs, and I have been here for a long time. I also find the fact disturbing the fact that you just make comments that you disagree as off-topic.

Anyway, CC @emilazy.

Reply:
@thiagokokada please stop re-opening this. you're polluting my PR with salt.

take a break, go for a walk, and if you then still think that I've wronged you, then take it to a channel, that's more appropriate. here, you're just making a fool of yourself.

Reply:
@bendlas Are you going to stop to ping me and hide my comments? If yes, ok, I will stop as long you also stop abusing your moderation powers.

Reply:
I really don’t want to get involved in this.

cc @NixOS/moderation

Reply:
Just for the record, and in case this comes up later:

- They started unresolving this thread, beginning with their first response https://github.com/NixOS/nixpkgs/pull/340663#discussion_r1760305136
- They stopped unresolving this thread only after I asked them to in https://github.com/NixOS/nixpkgs/pull/340663#discussion_r1760396722
  - I asked them to stop, because they started unresolving, without even adding messages
- I did mark https://github.com/NixOS/nixpkgs/pull/340663#discussion_r1760375104 as off-topic, which they reverted
  - this moderation action, I fully stand behind, as there was nothing in that message pertaining to this PR
- I did mark https://github.com/NixOS/nixpkgs/pull/340663#discussion_r1760391565 as spam, which they also reverted
  -  I did that, because they pinged a third party into the conversation, unsolicited, which qualifies as spam IMO; also because I wanted to make good on my promise to mute any further engagement on this thread
  -  still, I can't fully stand behind that moderation action, because they also said something pertaining to the PR in that message

In addition to the facts on the outer indentation, I added some of my reasoning one level in, so that I'm committed to it. Happy to give more reasoning to what happened here, but .. let's be honest .. who's actually trying to hear it 🥲

Reply:
I also gave multiple times my reason for unsolving this: the issue was not solved, OP itself said that this was going to backlog, other reviewers seem to concur this is a good idea (since the reaction), so there is no reason to solve unless the issue is actually solved OR there is a good explanation on why not to do so (that OP still didn't give a good explanation why not to, except for speed that I answered, some non-technical ones, and also in my opinion unnecessary harshly replies for a feedback).

I would definitely not escalate this or react as badly if OP didn't start implying that his decision to "solve" the issue is "respect" while my decision of trying to get an issue more visibility is "nitpicking". Marking messages as off-topic or spam was definitely the cherry on top, as far I know this always should be done by a third-party otherwise this is a good way to just make the whole discussion more heated.

Finally, I only pinged a third party because I didn't know we had a direct ping to the moderation team, so thanks `@emilylazy` for doing so.

Reply:
@thiagokokada please ... just stop now. Nobody's trying to hear it from you either and nobody is making you stay on this thread (and please don't try another "stop pinging me". That is ridiculous if you're just coming back after, even without being pinged)

I let you have your fun dogpiling on me in the other issue and refrained from responding there after [we had been asked to stop](https://github.com/NixOS/nixpkgs/pull/337478#issuecomment-2311494579). But this here is my PR! I'm trying to achieve something here.

I did ask you for feedback on this draft PR, but I was specific about what I needed right now (use cases). You ignored that and decided to open up multiple review threads about variable naming, config location and the shebang. The cost of those review threads is non-zero, especially if there are other conversations going on.

I did resolve your two items that I agreed with (variable naming, config location) in your favor, the shebang one I declined. Ultimately, my goal was to free up space in this PR for the conversations I'm actually looking for at the moment - use cases - some of which I'm thankfully getting as well.

You don't have a right to my attention, you don't have a right to my time, you don't have a right to visibility in this PR, or to have your pet issue addressed and ultimately [this whole thing isn't about you](https://gist.github.com/richhickey/1563cddea1002958f96e7ba9519972d9).

I can't stop you from making misrepresentations, or posting here at all. I can't make you respect me for trying to do custom application development for you. I can't stop you from trying to weaponize a dramatic exit. I can't make you read what I wrote and I can't make you respond to the points where I'm actually looking for feedback. I can't make you actually try out the script before offering feedback and I can't even make you try out your own suggestions before offering them.

But please, for the love of sanity, think for a moment, what you're trying to achieve here, before hitting the `Comment` button!

Reply:
> (and please don't try another "stop pinging me". That is ridiculous if you're just coming back after, even without being pinged)

I just unfollowed this specific thread (I didn't know how to do it before, sorry for my lack of knowledge even if I had specifically already unfollowed this PR) and now you ping me explicitly? Why? I asked to please not ping me.

> You don't have a right to my attention, you don't have a right to my time, you don't have a right to visibility in this PR, or to have your pet issue addressed and ultimately [this whole thing isn't about you](https://gist.github.com/richhickey/1563cddea1002958f96e7ba9519972d9).

Neither you have of my time.

> But please, for the love of sanity, think for a moment, what you're trying to achieve here, before hitting the Comment button!

That goes both ways.

Reply:
> now you ping me explicitly? Why? I asked to please not ping me.

still trying to get the last word in, hm? bad news for you: in my heart of hearts, I'm petty like that as well ... and you're still harrassing my PR

> BTW, just blocked you. You maybe should do the same with me

Good! I'm not blocking you, because I don't have a problem with you, as long as you don't harrass my PR.

Reply:
> This adds 1-2 seconds to the script startup, so no

This would matter in a CGI script but not the case. Self contained > faster.

Reply:
Hey, moderation team hat on. 

We're all humans, we're all have emotions. Let's all take that into account for a minute.

bendlas expressed he's mostly looking for overall feedback at this time. Debating over this shebang is unnecessarily heating the conversation for now. 

Let's stop getting back to this shebang point up until this PR gets in a ready state and gets undrafted. It'll help this conflict cool down a bit and improve the overall collaboration here.

Reply:
> This would matter in a CGI script but not the case.

I think it also matters in a CLI tool. Especially one that I'm hacking on.

> Self contained > faster.

I agree that ultimately, self-contained is pretty high up on the priority list, and I fully intend on making it so, before making this generally available.

However:

- there are more ways to do this, e.g. by compiling into a flake app, or other form of wrapper script. I have not yet decided what the best way will be, and it will depend on the use cases that might hopefully emerge
- in the current state, none of this is self-contained, even with a nix-shell shebang:
  - there are bits and pieces of functionality, that make little sense on their own
  - to get anything interesting, the scripts needs to be run multiple times and combined with other tools, e.g. see https://github.com/NixOS/nixpkgs/pull/340663#issuecomment-2351811711
  - adding the shebang at this point would give a false sense of self-containedness, that would be betrayed on first close inspection, leaving this whole effort arguably worse-off, than with forcing early users/reviewers to nix-shell themselves
  - the proposed change added `jq`, which is not used within the script, leaving me to assume that not much mental effort went into the proposal at all, rather purely mechanistic "ruling". I consider that kind of feedback to be detrimental during any exploratory coding

Still:

- Please don't consider resolution of this proposal as rejection. My motivation was and is, to keep the focus on exploration and for that, I needed to collapse the distracting threads.